stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java

```diff
@@ -269,14 +269,17 @@ private void processCommand(final LoggedEvent loggedEvent) {
         processingMetrics.observeCommandCount(processedCommandsCount);
       }
 
+      finalizeCommandProcessing();
+
       if (currentProcessingResult.isEmpty()) {
-        skipRecord();
+        updateState();
+
+        notifySkippedListener(currentRecord);
+        metrics.eventSkipped();
```

Comment:
:wrench: I think it's increasingly difficult to follow the control flow here. Since `writeRecords` already handles the case where there's nothing to write, we could let `writeRecords` also call `notifySkippedListener` and `eventSkipped` and remove the branch here. 
Then we'd always go though `processCommand` -> `writeRecords` -> `updateState`. That'd would also match closer to the state machine diagram at the top of this file.

Reply:
Sounds great let's do that

Reply:
@oleschoenburg like this https://github.com/camunda/zeebe/pull/13446/commits/c56578ba04f3d33edb57de9e32302a6da5e13b8c ?

Reply:
Yes, exactly :star_struck: